py/mattak/backends/pyroot/dataset.py

```diff
@@ -129,7 +129,10 @@ def _eventInfo(self, i : int) -> Optional[mattak.Dataset.EventInfo]:
         # the default value for the sampling rate (3.2 GHz) which is used
         # for data which does not contain this information in the waveform files
         # is set in the header fils Waveforms.h
-        sampleRate = self.ds.raw().radiant_sampling_rate / 1000
+        try:
+            sampleRate = self.ds.raw().radiant_sampling_rate / 1000
+        except ReferenceError:
+            sampleRate = None
```

Comment:
I'm not sure if this will cause bigger problems down the line?  Could just use a heuristic based on date (> 2024 = 2.4, < 2024 = 3.2)?

Reply:
I don't think this is a good idea. Really we just want to fix the bug, as I said it works fine with uproot (so presumably it infers the sampling rate of events with no waveforms from the events with waveforms). Your suggestion would not tell me that something had gone wrong, and it might lead to weird results such as different reported sampling rates for events with and without waveforms within a single run.

Reply:
well the problem is that the header itself doesn't have a sample rate stored with it, so unless you either have the waveforms or the run info, it won't work. It looks like the uproot backend just sets it to 3200 which I think is worse? https://github.com/RNO-G/mattak/blob/main/py/mattak/backends/uproot/dataset.py#L279

Reply:
having sample rate = None will potentially cause strange problems when someone tries to plot a spectrum

Reply:
I don't think this is true. E.g. 
```
import mattak.Dataset
ds = mattak.Dataset.Dataset(
    data_path="/home/sb/Python/scratch/data/inbox/station13/run240013/", 
    skip_incomplete=False, backend='uproot')
ds.setEntries(0)
ds.eventInfo().sampleRate
```
correctly returns `np.float64(2.4)`, whereas replacing `uproot` -> `pyroot` gives the above error again.

Reply:
> having sample rate = None will potentially cause strange problems when someone tries to plot a spectrum

Again, from an end-user perspective, I would much rather have code that raises an error than code that silently gives me the wrong answer (e.g. wrong spectra because the sampling rate is incorrect). As far as I understand @fschlueter implemented something that will return a sampling rate of 3.2 for files that do not have the field for sampling rate in `waveforms`, but I don't think it returns 3.2 if it does not find a waveform file.

Reply:
> I don't think this is true. E.g.
> 
> ```
> import mattak.Dataset
> ds = mattak.Dataset.Dataset(
>     data_path="/home/sb/Python/scratch/data/inbox/station13/run240013/", 
>     skip_incomplete=False, backend='uproot')
> ds.setEntries(0) # only header is present for this entry, no waveforms
> ds.eventInfo().sampleRate
> ```
> 
> correctly returns `np.float64(2.4)`, whereas replacing `uproot` -> `pyroot` gives the above error again.

Actually, having thought about this - it probably gets the correct answer for incorrect reasons, and indeed picking an index that is larger than the number of waveforms but smaller than the number of events raises an error with uproot also. Seeing as sampling rates shouldn't change mid-run, maybe the solution is to just always read the sampling rate from the first event in the waveforms file?

Reply:
I think that's reasonable, if the waveforms file is there. I guess in the case that no waveforms are present, the sample rate isn't useful for too much anyway so it's fine if it's None.